frontend/packages/dev-console/src/components/edit-deployment/deployment-strategy/DeploymentStrategySection.tsx

```diff
@@ -21,30 +23,59 @@ const DeploymentStrategySection: React.FC<DeploymentStrategySectionProps> = ({
   const { t } = useTranslation();
   const {
     values: {
-      deploymentStrategy: { type },
+      formData: {
+        name: resName,
+        project: { name: resNamespace },
+        deploymentStrategy: { type },
+      },
     },
+    initialValues,
+    setFieldValue,
   } = useFormikContext<FormikValues>();
 
   const deploymentStrategyFields = React.useMemo(() => {
     switch (type) {
-      case DeploymentStrategyType.recreate:
-        return <RecreateStrategy resourceType={resourceType} />;
-      case DeploymentStrategyType.rolling:
-        return <RollingStrategy resourceType={resourceType} />;
-      case DeploymentStrategyType.custom:
+      case DeploymentStrategyType.recreateParams:
+        return <RecreateStrategy resourceType={resourceType} resourceObj={resourceObj} />;
+      case DeploymentStrategyType.rollingParams:
+      case DeploymentStrategyType.rollingUpdate:
+        return <RollingStrategy resourceType={resourceType} resourceObj={resourceObj} />;
+      case DeploymentStrategyType.customParams:
         return <CustomStrategy resourceObj={resourceObj} />;
       default:
         return null;
     }
-  }, [resourceObj, resourceType, type]);
+  }, [type, resourceObj, resourceType]);
 
   return (
     <FormSection title={t('devconsole~Deployment strategy')}>
       <DropdownField
-        name="deploymentStrategy.type"
+        name="formData.deploymentStrategy.type"
         label={t('devconsole~Strategy type')}
         items={DeploymentStrategyDropdownData[resourceType].items}
         helpText={DeploymentStrategyDropdownData[resourceType].helpText[type]}
+        onChange={(value) => {
+          const strategyDefaultValues = getStrategyData(
+            value,
+            {},
+            resName,
+            resNamespace,
+            resourceType,
+          );
+          const strategyData = {
+            ...(_.omit(resourceObj.spec?.strategy, [
+              'rollingParams',
+              'recreateParams',
+              'customParams',
+              'rollingUpdate',
+            ]) ?? {}),
```

Comment:
Lodash omit always returns an object.
```suggestion
            ])),
```